telemetry/definitions/commonDefinitions.json

```diff
@@ -3294,6 +3438,419 @@
                 { "type": "codecatalyst_devEnvironmentWorkflowStep"},
                 { "type": "codecatalyst_devEnvironmentWorkflowError", "required": false }
             ]
+        },
+        {
+            "name": "codeTransform_isVisibleToUser",
+            "passive": true,
+            "description": "The code transform button will only be available to start transform when a Java module or build configuration is found in the repository.",
+            "metadata": [
+                {
+                    "type": "codeTransformSessionId",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "codeTransform_totalRunTime",
+            "passive": true,
+            "description": "The code transform button will only be available to start transform when a Java module or build configuration is found in the repository.",
+            "metadata": [
+                {
+                    "type": "codeTransformSessionId",
+                    "required": true
+                },
+                {
+                    "type": "codeTransformRunTimeLatency",
+                    "required": true
+                },
+                {
+                    "type": "codeTransformResultStatusMessage",
+                    "required": false
+                }
+            ]
+        },
+        {
+            "name": "codeTransform_isDoubleClickedToTriggerUserModal",
+            "description": "The code transform button in the devtools to initiate the pre transform user modal window.",
+            "metadata": [
+                {
+                    "type": "codeTransformSessionId",
+                    "required": true
+                },
+                {
+                    "type": "codeTransformStartSrcComponents",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "codeTransform_isDoubleClickedToTriggerInvalidProject",
+            "description": "The code transform button in the devtools to initiate the job but there is a validation error.",
+            "metadata": [
+                {
+                    "type": "codeTransformPreValidationError",
+                    "required": true
+                },
+                {
+                    "type": "codeTransformSessionId",
+                    "required": true
+                },
+                { "type": "result", "required": true },
```

Comment:
I don't think we need these two fields

Reply:
These were added by Richard, so I left them. The toolkit team uses `result` and `reason` for their own reporting so I want all backwards compatibility there.

Reply:
what does result capyure though?

Reply:
because if doubleClicking will always be success. but if result captures transform result then it makes sense. 
Maybe include in description what result means here

Reply:
This is for logging failures and "types" of failures. There are several ways a project validation can occur. Outlined by :

```
        {
            "name": "codeTransformPreValidationError",
            "type": "string",
            "description": "Names of the pre-validation errors that can occur in the project",
            "allowedValues": [
                "No pom.xml file found",
                "No Java project found",
                "Mixed Java project and another language found",
                "Project selected is not Java 8 or Java 11",
                "Only Maven projects supported",
                "Empty project",
                "Non SSO login",
                "Project running on backend"
            ]
        },
        ```